core/src/main/kotlin/pages/PageBuilder.kt

```diff
@@ -54,33 +62,86 @@ class DefaultPageBuilder(
         }
     }
 
-    private fun contentForClass(c: Class) = group(c) {
-        header(1) { text(c.name) }
-        c.inherited.takeIf { it.isNotEmpty() }?.let {
-            header(2) { text("SuperInterfaces") }
-            linkTable(it)
+    private fun contentForClasslike(c: Classlike): ContentGroup = group(c) {
+        headerForClasslike(c)
+
+        when (c) {
+            is Class -> partialContentForClass(c)
+            is Enum -> partialContentForEnum(c)
+            else -> TODO("$c should not be present here")
         }
-        c.commentsData.forEach {
+
+        functionsBlock(c)
+        propertiesBlock(c)
+    }
+
+    private fun partialContentForClass(c: Class) = group(c) {
+
+        bodyForClasslike(c)
+        block("Constructors", 2, ContentKind.Functions, c.constructors, c.platformData) {
+            link(it.name, it.dri)
+            signature(it)
+            text(it.briefDocTagString)
+        }
+    }
+
+    private fun partialContentForEnum(c: Enum) = group(c) {
+        contentForEnumEntries(c)
+
+        bodyForClasslike(c)
+        block("Constructors", 2, ContentKind.Functions, c.constructors, c.platformData) {
+            link(it.name, it.dri)
+            signature(it)
+            text(it.briefDocTagString)
+        }
+    }
+
+    private fun PageContentBuilder.headerForClasslike(c: Classlike) =
+        header(1) {
+            val prefix = when (c) {
+                is Class -> "class "
+                is Enum -> "enum "
+                else -> ""
+            }
+            text("$prefix${c.name}")
+        }
+
+    private fun PageContentBuilder.contentForEnumEntries(e: Enum) =
+        block("Entries", 2, ContentKind.Properties, e.entries, e.platformData) { entry ->
+            link(entry.name, entry.dri)
+            contentForComments(entry)
+        }
+
+    private fun PageContentBuilder.contentForComments(d: Documentable) =
+        d.commentsData.forEach {
             it.children.forEach {
                 header(3) { text(it.toHeaderString()) }
                 comment(it.root)
                 text("\n")
             }
         }
-        block("Constructors", 2, ContentKind.Functions, c.constructors, c.platformData) {
+
+    private fun PageContentBuilder.bodyForClasslike(c: Classlike) {
+        c.inherited.takeIf { it.isNotEmpty() }?.let {
+            header(2) { text("SuperInterfaces") }
+            linkTable(it)
+        }
+        contentForComments(c)
+    }
+
+    private fun PageContentBuilder.propertiesBlock(c: Classlike) {
+        block("Properties", 2, ContentKind.Properties, c.properties, c.platformData) {
             link(it.name, it.dri)
-            signature(it)
             text(it.briefDocTagString)
         }
+    }
+
+    private fun PageContentBuilder.functionsBlock(c: Classlike) {
```

Comment:
We shouldn't hack our way around `PageContentBuilder` like this. 
`contentForEnumEntries` is only needed once, so having it as a separate method seems like an overkill. 
Both `headerForClasslike, `contentForComments`, etc. should also be inlined. I'd suggest:

```
fun contentForClasslike(c: ClassLike) = when(c) {
  is Class -> contentForClass(c)
  is Enum -> contentForEnum(c)
  else -> throw AssertionError(or some better exception)
}
fun contentForClass(c: Class) = group(c) { 
  header(1) { "class" }
  block("Properties", 2, ContentKind.Properties, ... 
}
fun contentForEnum(e: Enum) = group(e) { 
  header(1) { "enum" }
  block("Properties", 2, ContentKind.Properties, ... 
}
``` 
and so on

Reply:
Changed.